File: Front/ProEventos-App/src/app/models/Palestrante.ts
import { Evento } from "./Evento";
import { UserUpdate } from "./identity/UserUpdate";
import { RedeSocial } from "./RedeSocial";

export interface Palestrante {
  id: number;
  minicurriculo: string;
  user: UserUpdate;
  redesSociais: RedeSocial[];
  palestrantesEventos: Evento[];
}

File: Back/src/ProEventos.Domain/Palestrante.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using ProEventos.Domain.Identity;

namespace ProEventos.Domain
{
    public class Palestrante
    {
        public int Id { get; set; }
        public string Minicurriculo { get; set; }
        public int UserId { get; set; }
        public User User { get; set; }
        public IEnumerable<RedeSocial> RedesSociais { get; set; }

        public IEnumerable<PalestranteEvento> PalestrantesEventos { get; set; }
    }
}